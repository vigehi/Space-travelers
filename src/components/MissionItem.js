import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMissions } from '../redux/missions/missions';

const MissionItem = (props) => {
  const { mission } = props;
  const dispatch = useDispatch();

  const handleClick = (e) => {
    dispatch(joinMissions({ id: e.target.id }));
  };

  return (
    <tr className="mission-item">
      <td className="mission-name">{mission.mission_name}</td>
      <td className="mission-desc">
        {mission.description}
        <span><a href={mission.more}>See more.</a></span>
      </td>
      <td className="t-align-center">
        {!mission.status && <span className="active-badge">NOT A MEMBER</span>}
        {mission.status && <span className="inactive-badge">Active Member</span>}
      </td>
      <td className="t-align-center">
        {!mission.status && <button id={mission.mission_id} onClick={handleClick} className="join-btn" type="button">Join Mission</button> }
        {mission.status && <button id={mission.mission_id} onClick={handleClick} className="leave-btn" type="button">Leave Mission</button>}
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  mission: PropTypes.shape({
    mission_name: PropTypes.string,
    description: PropTypes.string,
    status: PropTypes.bool,
    mission_id: PropTypes.string,
    more: PropTypes.string,
  }).isRequired,
};

export default MissionItem;
