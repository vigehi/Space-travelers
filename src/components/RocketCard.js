import React from 'react';
import propTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { rocketAction } from '../redux/rockets/rockets';

const RocketCard = (props) => {
  const dispatch = useDispatch();

  const {
    name, desc, src, value, id,
  } = props;

  const handler = (e) => {
    const { id } = e.target;
    dispatch(rocketAction.rocketReserve({ id }));
  };

  return (
    <div className="rocket-continer">
      <div className="img-continer">
        <img className="image" alt="roket-img" src={src} />
      </div>
      <div className="text-continer">
        <h1 className="rocket-name">{name}</h1>
        <div className="paragraph-cont">
          <p className="rocket-des">
            {' '}
            <span style={{ display: value ? 'inline' : 'none' }} className="reserved-button">Reserved</span>
            {' '}
            {desc}
          </p>
        </div>
        {!value && <button id={id} onClick={handler} className="card-button" type="button">Reserve Rocket</button> }
        {value && <button id={id} onClick={handler} className="cancel-button" type="button">Cancel Reservation</button>}
      </div>
    </div>
  );
};

export default RocketCard;

RocketCard.propTypes = {
  name: propTypes.string.isRequired,
  desc: propTypes.string.isRequired,
  src: propTypes.string.isRequired,
  value: propTypes.bool.isRequired,
  id: propTypes.string.isRequired,
};
