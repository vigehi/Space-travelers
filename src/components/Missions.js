import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import MissionItem from './MissionItem';
import { getMissions } from '../redux/missions/missions';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!missions.length) {
      dispatch(getMissions());
    }
  }, []);

  return (
    <div className="main-sec">
      <table id="missions-table">
        <thead>
          <tr>
            <th id="mission-col">Mission</th>
            <th id="description-col">Description</th>
            <th id="status-col">Status</th>
            <th> </th>
          </tr>
        </thead>

        <tbody>
          {missions.map((mission) => (
            <MissionItem mission={mission} key={mission.mission_id} />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
