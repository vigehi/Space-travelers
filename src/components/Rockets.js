import React from 'react';
import { useSelector } from 'react-redux';
import RocketCard from './RocketCard';

const Rockets = () => {
  const state = useSelector((state) => state.storeSlice);

  return (
    <>
      <div>
        {state.map((rocket) => (
          <RocketCard
            key={rocket.id}
            id={rocket.id}
            name={rocket.name}
            src={rocket.img}
            desc={rocket.description}
            value={rocket.value}
          />
        ))}

      </div>
    </>
  );
};

export default Rockets;
