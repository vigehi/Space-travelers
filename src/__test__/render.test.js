import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Rockets from '../components/Rockets';
import Profile from '../components/profile';
import Missions from '../components/Missions';
import store from '../redux/configureStore';
import MissionItem from '../components/MissionItem';
import RocketCard from '../components/RocketCard';
import Navbar from '../components/navbar';
import ProfileCard from '../components/profileCard';

describe('Space App components rendering test', () => {
  it('match snapshot for NavBar', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for Rockets', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Rockets />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for Rocket Card', () => {
    const rocket = {
      id: 'Rocket 1',
      name: 'Rocket 1',
      description: 'This is the description for mission 1',
      img: 'a link',
      value: false,
    };

    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <RocketCard
            key={rocket.id}
            id={rocket.id}
            name={rocket.name}
            src={rocket.img}
            desc={rocket.description}
            value={rocket.value}
          />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for the Missions', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Missions />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for the Mission Item', () => {
    const mission = {
      mission_name: 'Mission 1',
      description: 'This is the description for mission 1',
      more: 'A link',
      status: false,
    };

    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <MissionItem mission={mission} />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for the Profile', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Profile />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for the Profile List', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <ProfileCard />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
