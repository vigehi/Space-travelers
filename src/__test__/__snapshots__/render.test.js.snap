// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space App components rendering test match snapshot for NavBar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="Nav-bar"
      >
        <div
          class="icon-continer"
        >
          <img
            alt="logo-img"
            class="logo-image"
            src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
          />
          <h1
            class="nav-text"
          >
            Space Traveler's Hub
          </h1>
        </div>
        <div
          class="options-continer"
        >
          <li>
            <a
              aria-current="page"
              class="Nav-bar-element active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="Nav-bar-element"
              href="/Dragons"
            >
              Dragons
            </a>
          </li>
          <li>
            <a
              class="Nav-bar-element"
              href="/Missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="Nav-bar-element"
              href="/MyProfile"
            >
              My Profile
            </a>
          </li>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="Nav-bar"
    >
      <div
        class="icon-continer"
      >
        <img
          alt="logo-img"
          class="logo-image"
          src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
        />
        <h1
          class="nav-text"
        >
          Space Traveler's Hub
        </h1>
      </div>
      <div
        class="options-continer"
      >
        <li>
          <a
            aria-current="page"
            class="Nav-bar-element active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="Nav-bar-element"
            href="/Dragons"
          >
            Dragons
          </a>
        </li>
        <li>
          <a
            class="Nav-bar-element"
            href="/Missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="Nav-bar-element"
            href="/MyProfile"
          >
            My Profile
          </a>
        </li>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for Rocket Card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket-continer"
      >
        <div
          class="img-continer"
        >
          <img
            alt="roket-img"
            class="image"
            src="a link"
          />
        </div>
        <div
          class="text-continer"
        >
          <h1
            class="rocket-name"
          >
            Rocket 1
          </h1>
          <div
            class="paragraph-cont"
          >
            <p
              class="rocket-des"
            >
               
              <span
                class="reserved-button"
                style="display: none;"
              >
                Reserved
              </span>
               
              This is the description for mission 1
            </p>
          </div>
          <button
            class="card-button"
            id="Rocket 1"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket-continer"
    >
      <div
        class="img-continer"
      >
        <img
          alt="roket-img"
          class="image"
          src="a link"
        />
      </div>
      <div
        class="text-continer"
      >
        <h1
          class="rocket-name"
        >
          Rocket 1
        </h1>
        <div
          class="paragraph-cont"
        >
          <p
            class="rocket-des"
          >
             
            <span
              class="reserved-button"
              style="display: none;"
            >
              Reserved
            </span>
             
            This is the description for mission 1
          </p>
        </div>
        <button
          class="card-button"
          id="Rocket 1"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for Rockets 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
    </div>
  </body>,
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for the Footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        id="footer"
      >
        <p>
          Microverse © 2022 All rights reserved.
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      id="footer"
    >
      <p>
        Microverse © 2022 All rights reserved.
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for the Mission Item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="mission-item"
      >
        <td
          class="mission-name"
        >
          Mission 1
        </td>
        <td
          class="mission-desc"
        >
          This is the description for mission 1
          <span>
            <a
              href="A link"
            >
              See more.
            </a>
          </span>
        </td>
        <td
          class="t-align-center"
        >
          <span
            class="active-badge"
          >
            NOT A MEMBER
          </span>
        </td>
        <td
          class="t-align-center"
        >
          <button
            class="join-btn"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="mission-item"
    >
      <td
        class="mission-name"
      >
        Mission 1
      </td>
      <td
        class="mission-desc"
      >
        This is the description for mission 1
        <span>
          <a
            href="A link"
          >
            See more.
          </a>
        </span>
      </td>
      <td
        class="t-align-center"
      >
        <span
          class="active-badge"
        >
          NOT A MEMBER
        </span>
      </td>
      <td
        class="t-align-center"
      >
        <button
          class="join-btn"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for the Missions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-sec"
      >
        <table
          id="missions-table"
        >
          <thead>
            <tr>
              <th
                id="mission-col"
              >
                Mission
              </th>
              <th
                id="description-col"
              >
                Description
              </th>
              <th
                id="status-col"
              >
                Status
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-sec"
    >
      <table
        id="missions-table"
      >
        <thead>
          <tr>
            <th
              id="mission-col"
            >
              Mission
            </th>
            <th
              id="description-col"
            >
              Description
            </th>
            <th
              id="status-col"
            >
              Status
            </th>
            <th>
               
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for the Profile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-cont"
      >
        <div
          class="user-reserved"
        >
          <h2>
            My Missions
            <span>
              🕹
            </span>
          </h2>
          <div
            id="empty-msg"
          >
            <span>
              No missions joined
            </span>
          </div>
          <a
            href="/Missions"
          >
            <button
              class="join-reserv-profile"
              type="button"
            >
              Join a Mission
            </button>
          </a>
        </div>
        <div
          class="user-reserved"
        >
          <h2>
            My Rockets
             
            <span>
              🚀
            </span>
          </h2>
          <div
            id="empty-msg"
          >
            <span>
              No rockets reserved
            </span>
          </div>
          <a
            href="/"
          >
            <button
              class="join-reserv-profile"
              type="button"
            >
              Reserve a rocket
            </button>
          </a>
        </div>
        <div
          class="user-reserved"
        >
          <h2>
            My Dragons
             
            <span>
              🚅
            </span>
          </h2>
          <div
            id="empty-msg"
          >
            <span>
              No dragons reserved
            </span>
          </div>
          <a
            href="/Dragons"
          >
            <button
              class="join-reserv-profile"
              type="button"
            >
              Reserve a dragon
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-cont"
    >
      <div
        class="user-reserved"
      >
        <h2>
          My Missions
          <span>
            🕹
          </span>
        </h2>
        <div
          id="empty-msg"
        >
          <span>
            No missions joined
          </span>
        </div>
        <a
          href="/Missions"
        >
          <button
            class="join-reserv-profile"
            type="button"
          >
            Join a Mission
          </button>
        </a>
      </div>
      <div
        class="user-reserved"
      >
        <h2>
          My Rockets
           
          <span>
            🚀
          </span>
        </h2>
        <div
          id="empty-msg"
        >
          <span>
            No rockets reserved
          </span>
        </div>
        <a
          href="/"
        >
          <button
            class="join-reserv-profile"
            type="button"
          >
            Reserve a rocket
          </button>
        </a>
      </div>
      <div
        class="user-reserved"
      >
        <h2>
          My Dragons
           
          <span>
            🚅
          </span>
        </h2>
        <div
          id="empty-msg"
        >
          <span>
            No dragons reserved
          </span>
        </div>
        <a
          href="/Dragons"
        >
          <button
            class="join-reserv-profile"
            type="button"
          >
            Reserve a dragon
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Space App components rendering test match snapshot for the Profile List 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-card"
      >
        <h3>
          default
        </h3>
        <div
          class="btn-container"
        >
          <button
            type="button"
          />
          <a
            class="learn-more"
          >
            Learn more
            <span
              class="arrow-more"
            >
              ➡
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-card"
    >
      <h3>
        default
      </h3>
      <div
        class="btn-container"
      >
        <button
          type="button"
        />
        <a
          class="learn-more"
        >
          Learn more
          <span
            class="arrow-more"
          >
            ➡
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
